### PR TITLE
[function](bitmap) support bitmap_to_base64 and bitmap_from_base64

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jacktengg** [comment]
run buildall

File: be/src/vec/functions/function_bitmap.cpp [hunk 1]
```diff
@@ -887,6 +942,40 @@ struct BitmapToString {
...
+    using Offsets = ColumnString::Offsets;
+
+    static Status vector(const std::vector<BitmapValue>& data, Chars& chars, Offsets& offsets) {
+        size_t size = data.size();
+        offsets.resize(size);
+        chars.reserve(size);
+        for (size_t i = 0; i < size; ++i) {
+            BitmapValue& bitmap_val = const_cast<BitmapValue&>(data[i]);
+            auto ser_size = bitmap_val.getSizeInBytes();
+            char ser_buff[ser_size];
```
Comment (mrhhsg) [inline be/src/vec/functions/function_bitmap.cpp:964]: Better to allocate this buffer on the heap memory and outside of the loop, and reuse it.

**Review by mrhhsg — CHANGES_REQUESTED** [review]

File: be/src/vec/functions/function_bitmap.cpp (hunk 1 above)
Comment (yiguolei) [inline be/src/vec/functions/function_bitmap.cpp:964]: not allocate on stack, the value maybe very large. allocate on heap.

File: be/src/vec/functions/function_bitmap.cpp [hunk 2]
```diff
@@ -887,6 +942,40 @@ struct BitmapToString {
...
+            bitmap_val.write_to(ser_buff);
+
+            int encoded_size = (int)(4.0 * ceil((double)ser_size / 3.0));
+            char dst[encoded_size];
+            memset(dst, 0, encoded_size);
+            int outlen =
+                    base64_encode((const unsigned char*)ser_buff, ser_size, (unsigned char*)dst);
+            DCHECK(outlen > 0);
+            StringOP::push_value_string(std::string_view(dst, outlen), i, chars, offsets);
```
Comment (yiguolei) [inline be/src/vec/functions/function_bitmap.cpp:973]: IF the output string will copy the column, then we could reuse the buffer. And realloc a new buffer if it size is less than needed.

File: be/src/vec/functions/function_bitmap.cpp [hunk 3]
```diff
@@ -250,6 +252,59 @@ struct BitmapFromString {
...
+            if (0 != src_size % 4) {
+                // return Status::InvalidArgument(
+                //         fmt::format("invalid base64: {}", std::string(src_str, src_size)));
+                res.emplace_back();
+                null_map[i] = 1;
+                continue;
+            }
+            curr_decode_buff_len = src_size + 3;
+            if (curr_decode_buff_len > last_decode_buff_len) {
+                decode_buff.reset(new char[curr_decode_buff_len]);
```
Comment (yiguolei) [inline be/src/vec/functions/function_bitmap.cpp:288]: reuse the buffer

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.7 seconds
 stream load tsv:          535 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17162175283 Bytes

**Comment by jacktengg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.23 seconds
 stream load tsv:          533 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162143293 Bytes

**Review by mrhhsg — APPROVED** [review]
lgtm

**Pull request merged**